a4kSubtitles/search.py

```diff
@@ -227,22 +227,63 @@ def __prepare_results(core, meta, results):
                 color_list = group
                 break
 
+    def _filter_name(x):
+        name_diff_ignore = quality + codec + audio + color
+        name_diff_ignore += ["multi", 'multiple', 'sub', 'subs', 'subtitle']
+
+        if x.isdigit():
+            x = str(int(x)).zfill(3)
+        elif x.lower() in name_diff_ignore:
+            x = ''
+        return x.lower()
+
+    def _match_numbers(a, b):
+        offset = 0
+        for s in b:
+            s = core.re.sub(r'v[1-4]', "", s)
+            if s.isdigit() and meta.episode and s.zfill(3) == meta.episode.zfill(3):
+                offset += 0.4
+            elif s.isdigit() and s in a:
+                offset += 0.2
+
+        return offset
+
     def sorter(x):
         name = x['name'].lower()
         nameparts = core.re.split(regexsplitwords, name)
 
+        cleaned_nameparts = list(filter(len, map(_filter_name, nameparts)))
+        cleaned_file_nameparts = list(filter(len, map(_filter_name, meta_nameparts)))
+        found_season, found_episode = core.utils.extract_season_episode(name)
+
+        incorrect_sub_offset = 0
+        is_season = found_season and found_season == meta.season.zfill(3)
+        is_episode = found_episode and found_episode == meta.episode.zfill(3)
+
+        # Handle the parsed season and episode.
+        if is_season and not found_episode:
+            incorrect_sub_offset += 0.6
+        if is_season and is_episode:
+            incorrect_sub_offset += 0.4
+        elif found_season and found_episode:
+            incorrect_sub_offset -= 0.5
+
+        if meta.is_tvshow and incorrect_sub_offset == 0:
```

Comment:
Move all the season and episode specific logic behind an `if meta.is_tvshow:` check to avoid causing issues with movies.